server/src/main/java/io/spine/server/reflect/RejectionHandlerMethod.java

```diff
@@ -0,0 +1,242 @@
+        final Class[] paramTypes = method.getParameterTypes();
+        final int paramCount = paramTypes.length;
+        switch (paramCount) {
+            case 1:
+                return Kind.REJECTION_MESSAGE_AWARE;
+            case 2:
+                final Class<?> secondParamType = paramTypes[1];
+                return  secondParamType == CommandContext.class
+                        ? Kind.COMMAND_CONTEXT_AWARE
+                        : Kind.COMMAND_MESSAGE_AWARE;
+            case 3:
+                return Kind.COMMAND_AWARE;
+            default:
+                throw newIllegalArgumentException(
+                        "Invalid Rejection handler method parameter count: %s.", paramCount);
+        }
+    }
+
+    /**
+     * Invokes the wrapped handler method to handle {@code rejectionMessage},
+     * {@code commandMessage} with the passed {@code context} of the {@code Command}.
+     *
+     * <p>Unlike the {@linkplain #invoke(Object, Message, Message) overloaded alternative method},
+     * this one may return some value.
+     *
+     * @param target       the target object on which call the method
+     * @param rejectionMsg the rejection message to handle
+     * @param commandMsg   the command message
+     * @param ctx          the context of the command
+     * @return the result of the invocation
+     */
+    Object doInvoke(Object target, Message rejectionMsg, Message commandMsg, CommandContext ctx) {
+        checkNotNull(rejectionMsg);
+        checkNotNull(commandMsg);
+        checkNotNull(ctx);
+
+        try {
+            final Object output;
+            final Method method = getMethod();
+            switch (kind) {
+                case REJECTION_MESSAGE_AWARE:
+                    output = method.invoke(target, rejectionMsg);
+                    break;
+                case COMMAND_CONTEXT_AWARE:
+                    output = method.invoke(target, rejectionMsg, ctx);
+                    break;
+                case COMMAND_MESSAGE_AWARE:
+                    output = method.invoke(target, rejectionMsg, commandMsg);
+                    break;
+                case COMMAND_AWARE:
+                    output = method.invoke(target, rejectionMsg, commandMsg, ctx);
+                    break;
+                default:
+                    throw unsupported("Unsupported method kind encountered %s", kind.name());
+            }
+            return output;
+        } catch (IllegalArgumentException | IllegalAccessException | InvocationTargetException e) {
+            throw whyFailed(target, rejectionMsg, ctx, e);
+        }
+    }
+
+    static IllegalStateException missingRejectionHandler(
+            Class<?> cls, Class<? extends Message> rejectionClass) {
+        final String msg = format(
+                "Missing handler for rejection class %s in the class %s",
+                rejectionClass, cls
+        );
+        return new IllegalStateException(msg);
```

Comment:
Using `newIllegalStateException()` which can format a string would make the above code more compact.